src/frontend/src/content/docs/integrations/frameworks/javascript.mdx

```diff
@@ -0,0 +1,265 @@
+
+The integration exposes a number of app resource types:
+
+- `JavaScriptAppResource`: Added with the `AddJavaScriptApp` method for general JavaScript applications
+- `NodeAppResource`: Added with the `AddNodeApp` method for running specific JavaScript files with Node.js
+- `ViteAppResource`: Added with the `AddViteApp` method for Vite applications with Vite-specific defaults
+
+## Add JavaScript application
+
+The `AddJavaScriptApp` method is the foundational method for adding JavaScript applications to your Aspire AppHost. It provides a consistent way to orchestrate JavaScript applications with automatic package manager detection and intelligent defaults.
+
+```csharp title="C# — AppHost.cs"
+var builder = DistributedApplication.CreateBuilder(args);
+
+var frontend = builder.AddJavaScriptApp("frontend", "./frontend")
+    .WithHttpEndpoint(port: 3000, env: "PORT");
+
+builder.AddProject<Projects.ExampleProject>()
+    .WithReference(frontend);
+
+// After adding all resources, run the app...
+```
+
+By default, `AddJavaScriptApp`:
+- Uses npm as the package manager when `package.json` is present
+- Runs the "dev" script during local development
+- Runs the "build" script when publishing to create production assets
+- Automatically generates Dockerfiles to build production assets
+
+The method accepts the following parameters:
+- `name`: The name of the resource in the Aspire dashboard
+- `workingDirectory`: The path to the directory containing your JavaScript application (where `package.json` is located)
+
+## Add Node.js application
+
+For Node.js applications that don't use a package.json script runner, you can directly run a JavaScript file using the `AddNodeApp` extension method:
+
+```csharp title="C# — AppHost.cs"
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nodeApp = builder.AddNodeApp("node-app", "server.js")
+    .WithHttpEndpoint(port: 3000, env: "PORT");
+
+builder.AddProject<Projects.ExampleProject>()
+    .WithReference(nodeApp);
+
+// After adding all resources, run the app...
+```
+
+The `AddNodeApp` method requires:
+- **name**: The name of the resource in the Aspire dashboard
+- **scriptPath**: The path to the JavaScript file to execute
+- **workingDirectory** (optional): The working directory (defaults to the script's directory)
+- **args** (optional): Command-line arguments to pass to the script
+
+## Add Vite application
+
+For Vite applications, you can use the `AddViteApp` extension method which provides Vite-specific defaults and optimizations:
+
+```csharp title="C# — AppHost.cs"
+var builder = DistributedApplication.CreateBuilder(args);
+
+var viteApp = builder.AddViteApp("vite-app", "./vite-app")
+    .WithHttpEndpoint(port: 5173, env: "PORT");
+
+builder.AddProject<Projects.ExampleProject>()
+    .WithReference(viteApp);
+
+// After adding all resources, run the app...
+```
+
+`AddViteApp` automatically configures:
+- **Development**: Runs the "dev" script (typically `vite`)
```

Comment:
What does `**Development**` mean here?

Reply:
It's the default script that is ran, so it's local `npm run dev`. Maybe we should make that more clear?